backend/timebank/src/main/kotlin/kookmin/software/capstone2023/timebank/application/service/inqui/InquiryService.kt

```diff
@@ -0,0 +1,138 @@
+     */
+    data class InquiryCreateRequest(
+            val title: String,
+            val content: String,
+            val userId: Long,
+            val inquiryDate: LocalDateTime = LocalDateTime.now()
+    )
+
+    /**
+     * 답변 Dto
+     */
+    data class InquiryUpdateRequest(
+            val replyContent: String?,
+            val replyStatus: InquiryStatus?,
+            val replyDate: LocalDateTime? = LocalDateTime.now()
+    )
+
+    /**
+     * 서비스 메소드 정의
+     */
+
+    /**
+     * 문의 생성 service
+     */
+    @Transactional
+    fun createInquiry(request: InquiryCreateRequest): InquiryDto {
+        val user = userJpaRepository.findById(request.userId)
+                .orElseThrow { IllegalArgumentException("User not found with id: ${request.userId}") }
+        val inquiry = Inquiry(title = request.title, content = request.content, user = user, inquiryDate = request.inquiryDate)
+        val savedInquiry = inquiryRepository.save(inquiry)
+        return inquiryToDto(savedInquiry)
+    }
+
+    /**
+     * 전체 문의 검색 service
+     */
+    fun getInquiries(): List<InquiryDto> {
+        val inquiries = inquiryRepository.findAll()
+        return inquiries.map(::inquiryToDto)
+    }
+
+    /**
+     * 문의ID검색 service
+     */
+    fun getInquiryById(id: Long): InquiryDto {
+        val inquiry = inquiryRepository.findById(id)
+                .orElseThrow { IllegalArgumentException("Inquiry not found with id: $id") }
+        return inquiryToDto(inquiry)
+    }
+
+    /**
+     * userId 검색 service
+     */
+    fun getInquiriesByUserId(userId: Long): List<InquiryDto> {
+        val user = userJpaRepository.findById(userId).orElseThrow { IllegalArgumentException("User not found with id: $userId") }
+        val inquiries = inquiryRepository.findByUser(user)
+        return inquiries.map { inquiry -> inquiryToDto(inquiry) }
+    }
+
+    /**
+     * 답변 service
+     */
+    fun updateInquiry(id: Long, request: InquiryUpdateRequest): InquiryDto {
+        val inquiry = inquiryRepository.findById(id)
+                .orElseThrow { IllegalArgumentException("Inquiry not found with id: $id") }
+        inquiry.replyContent = request.replyContent ?: inquiry.replyContent
+        inquiry.replyStatus = request.replyStatus ?: inquiry.replyStatus
+        inquiry.replyDate = request.replyDate ?: inquiry.replyDate
+        val updatedInquiry = inquiryRepository.save(inquiry)
+        return inquiryToDto(updatedInquiry)
+    }
+
+    /**
+     * 문의삭제 service
+     */
+    fun deleteInquiryByUserId(userId: Long, inquiryId: Long) {
+        val user = userJpaRepository.findById(userId).orElseThrow { IllegalArgumentException("User not found with id: $userId") }
+        val inquiry = inquiryRepository.findById(inquiryId).orElseThrow { IllegalArgumentException("Inquiry not found with id: $inquiryId") }
+
+        if (inquiry.user.id != user.id) {
+            throw IllegalArgumentException("User does not have permission to delete this inquiry")
```

Comment:
```suggestion
            throw UnauthorizedException("User does not have permission to delete this inquiry")
```
미리 정의해둔 Exception을 사용하면 좋을 것 같아요~